helpers/progress.go

```diff
@@ -10,16 +10,30 @@ func (DiscardProgressWriter) Write(p []byte) (int, error) {
 	return len(p), nil
 }
 
-// UpdateTitle doesn't do anything but satisfy implementation
-func (DiscardProgressWriter) UpdateTitle(_ string) {}
+// Close doesn't do anything but satisfy implementation
+func (DiscardProgressWriter) Close() error {
+	return nil
+}
+
+// Updatef doesn't do anything but satisfy implementation
+func (DiscardProgressWriter) Updatef(_ string, _ ...any) {}
+
+// Successf doesn't do anything but satisfy implementation
+func (DiscardProgressWriter) Successf(_ string, _ ...any) {}
+
+// Failf doesn't do anything but satisfy implementation
+func (DiscardProgressWriter) Failf(_ string, _ ...any) {}
 
 // DiscardProgressWriter is a ProgressWriter in which all calls succeed without doing anything
 // Use this or nil or if you don't care about writing progress
 type DiscardProgressWriter struct{}
 
-// ProgressWriter wraps io.Writer, but also includes an updateTitle function to give the user
+// ProgressWriter wraps io.Writer, but also includes functions to give the user
 // additional context on what's going on. Useful in OCI for tracking layers
 type ProgressWriter interface {
-	UpdateTitle(string)
+	Updatef(string, ...any)
+	Successf(string, ...any)
+	Failf(string, ...any)
```

Comment:
This is likely not a significant issue, but curious what the pros/cons of passing in an `error` object as a the first argument to `Failf` are?

We currently use this signature in Zarf, but we might not lose much by getting rid of that argument:

https://github.com/defenseunicorns/zarf/blob/32b6c2473d365cc0c6eccfff130152f56eb58703/src/pkg/message/progress.go#L97-L101

Reply:
My thoughts are that the signature with an error is a bit overloaded and it isn't too hard to handle that with a quick call to message.Debug() if you want to handle it that way (there isn't really too much special going on with the error object in the function)